.github/workflows/soak.yml

```diff
@@ -0,0 +1,244 @@
+name: Soak
+
+on: push
+
+jobs:
+  cancel-previous:
+    runs-on: ubuntu-20.04
+    timeout-minutes: 3
+    if: github.ref != 'refs/heads/master'
+    steps:
+      - uses: styfle/cancel-workflow-action@0.9.1
+        with:
+          access_token: ${{ secrets.GITHUB_TOKEN }}
+          all_but_latest: true # can cancel workflows scheduled later
+
+  build-baseline-image:
+    name: Build baseline 'soak-vector' container
+    runs-on: [self-hosted, linux, x64, general]
+    steps:
+      - uses: actions/checkout@v2.3.5
+
+      - uses: actions/checkout@v2.3.5
+        with:
+          ref: master
+          path: baseline-vector
+
+      - name: Set baseline SHA, store in file
+        id: baseline
+        run: |
+          echo ::set-output name=SHA::$(git rev-parse HEAD)
+          git rev-parse HEAD > sha
```

Comment:
Wouldn't we want to execute this in the `baseline-vector` directory?

Reply:
Hmm, good point, and yet this appears to function correctly.

Reply:
It only appeared to. Resolved in 4d9a6283fae9cc2a4511b75e29cd1fa18321f351